src/application.rs

```diff
@@ -119,6 +121,23 @@ impl Application {
         }));
         self.add_action(&action_about);
 
+        // Mark message as viewed
+        let action_mark_message_as_viewed = gio::SimpleAction::new(
+            "mark-message-as-viewed",
+            Some(glib::VariantTy::new("(ixx)").unwrap()),
+        );
+        action_mark_message_as_viewed.connect_activate(
+            clone!(@weak self as app => move |_, data| {
+                let (client_id, chat_id, message_id) = data.unwrap().get().unwrap();
+                spawn(async move {
+                    functions::view_messages(chat_id, 0, vec![message_id], true, client_id)
+                        .await
+                        .unwrap();
```

Comment:
The question is whether we use this (just marking the message in the notification as viewed) or viewing the whole chat (https://github.com/melix99/telegrand/blob/8258df00b20f07b6b637349949f514e3bdfdce9d/src/tdlib/chat.rs#L655). Tdesktop seems to do the latter.